### ISSUE TITLE
LDAP Groups: solution in #19970 solves visible, but ...

### ISSUE THREAD

**Issue by dea-75** [nextcloud/server#20133]
Now, using the solution in #19970 NEW LDAP groups in Nextcloud 18.0.3 are visible.
But if I try to share a directory to a NEW LDAP GROUP, the share is activated, but users in the LDAP group don’t see the share.

If I try to share directly to LDAP user (no the LDAP GROUP), works fine.

Is the problem related?

**Comment by kesselb** [comment]
Thank you for reporting this issue :+1: 

Would you mind to edit your post and add the [issue template](https://github.com/nextcloud/server/blob/master/.github/ISSUE_TEMPLATE/Bug_report.md)? The information are required to reproduce your issue.

**Comment by dea-75** [comment]
Steps to reproduce
Nextcloud 18.0.3, only LDAP users and groups. Apply patch in #19970. Now LDAP groups are visible.
Expected behaviour
When I share a folder to a new added LDAP group, share is activated, but users in the LDAP group doesn't see the share. If I share to singles LDAP users works fine (the problem is present ONLY for LDAP groups).
If I create a local groups (to test) and insert several LDAP users in the local group, works fine.

NO APP "Group folders" installed.

OS: Debian 10
Apache 2.4 (Debian)
PHP 7.3 - FPM (Debian 10)

Nextcloud 18.0.3 with #19970 applied
LDAP users app installed, last version.

No extenal storage, No encryption.

**Comment by dea-75** [comment]
Sorry, it seems to work, I can no longer reproduce the non-sharing.
Very thanks for the support !!

**Comment by MatzeFix** [comment]
Hey together,

we have the same system as dea-75 -> also Version 18.03 and the same system-versions.
We also have the same problem.

@dea-75 did you make a workaround or some steps to fix that?
I tested this problem in Version 18.00; 18.02 and 18.03 and got always the same result.
Sharing folders with LDAP groups don't work (like dea-75 discribed).
We tested the LDAP connection and the results of the requests. All works perfekt, but we can't share folders with LDAP groups.
I don't know what to do, to fix it.... I also tested it in the past with Group-Folders (this works perfect).

**Comment by MatzeFix** [comment]
Hey together,

I found a workaround to the problem - which is not pretty nice but it works.

Actual status:
At the moment, it's possible to to share folders to LDAP-Groups. But the LDAP-Group Members don't see the shared folder.

Workaround:
In the database there is a table called oc_share which organize all shared folders with names etc.
This table has the attribute "accepted". By default this entry has the value 0. If you change all entries with accepted = 0 to accepted = 1 -> the shared folders and files will be avaliable for the LDAP-Group members on their own root-directory.

I tested this workaround for 18.02 and 18.03 and it works.
Maybe the feature -> accept shared folders an document has a bug.

We also made some config changes with the option following options to true and false (also with relogin after change an so on), but it doesn't change the problem.
 * Set to true to enable that internal shares need to be accepted by the users $
 * Users can change this for their account in their personal sharing settings
 */
'sharing.enable_share_accept' => false,
/**
 * Set to true to enforce that internal shares need to be accepted
 */
'sharing.force_share_accept' => false,
